youtubeBot/bot.py

```diff
@@ -32,8 +32,28 @@ def send_text(self, text):
 
 
 class YoutubeBot(Bot):
-    pass  # TODO your code here!
 
+    def __init__(self, token):
+        self.ctr = 0
+        self.videos = []
+        super().__init__(token)
+
+    def message_handler(self, message):
+        if(self.ctr == 0):
+             super().send_text('Downloading ' + message)
+             self.videos = download_youtube_file(message, 2)
+             super().send_text(self.videos)
+             super().send_text('choose video index')
+             self.ctr += 1
+
+        else:
+            if(int(message) < len(self.videos)):
+                super().send_text('sending')
+                super().send_video(self.videos[int(message)])
+                self.ctr = 0
+            else:
+                super().send_text('pick another index')
```

Comment:
Great solution. 
Note that you should use "self" instead "suprer()" since  YoutubeBot inherits from Bot